packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardTitle.base.tsx

```diff
@@ -9,66 +9,65 @@ const getClassNames = classNamesFunction<IDocumentCardTitleStyleProps, IDocument
 export interface IDocumentCardTitleState {
   truncatedTitleFirstPiece?: string;
   truncatedTitleSecondPiece?: string;
+  clientWidth?: number;
+  previousTitle: string;
+  needMeasurement: boolean;
 }
 
-const TRUNCATION_SEPARATOR = '&hellip;';
-const TRUNCATION_MINIMUM_LENGTH = 40; // This is the length we know can fit into the min width of DocumentCard.
-const TRUNCATION_MAXIMUM_LENGTH = 90 - TRUNCATION_SEPARATOR.length;
-
-// This is the length we know can fit into the min width 2 lines of DocumentCard.
-const TRUNCATION_MINI_LENGTH_SECONDARY = 80;
-const TRUNCATION_MAX_LENGTH_SECONDARY = 130 - TRUNCATION_SEPARATOR.length;
-const TRUNCATION_FIRST_PIECE_LONGER_BY = 10;
 const TRUNCATION_VERTICAL_OVERFLOW_THRESHOLD = 5;
 
 export class DocumentCardTitleBase extends BaseComponent<IDocumentCardTitleProps, IDocumentCardTitleState> {
   private _titleElement = React.createRef<HTMLDivElement>();
+  private _measureTitleElement = React.createRef<HTMLDivElement>();
+
   private _scrollTimerId: number;
-  private _truncatedTitleAtWidth: number;
-  private _isTruncated: boolean;
   private _classNames: IProcessedStyleSet<IDocumentCardTitleStyles>;
 
   constructor(props: IDocumentCardTitleProps) {
     super(props);
 
     this.state = {
       truncatedTitleFirstPiece: '',
-      truncatedTitleSecondPiece: ''
+      truncatedTitleSecondPiece: '',
+      previousTitle: props.title,
+      needMeasurement: !!props.shouldTruncate
     };
   }
 
-  public componentDidMount(): void {
-    const { title, shouldTruncate, showAsSecondaryTitle } = this.props;
-    const miniLength = showAsSecondaryTitle ? TRUNCATION_MINI_LENGTH_SECONDARY : TRUNCATION_MINIMUM_LENGTH;
-    if (shouldTruncate && title && title.length > miniLength) {
-      if (this._doesTitleOverflow()) {
-        this._startTruncation(this.props);
-      }
-      this._events.on(window, 'resize', this._updateTruncation);
+  public componentDidUpdate(): void {
+    if (this.props.title !== this.state.previousTitle) {
+      this.setState({
+        truncatedTitleFirstPiece: undefined,
+        truncatedTitleSecondPiece: undefined,
+        clientWidth: undefined,
+        previousTitle: this.props.title,
+        needMeasurement: !!this.props.shouldTruncate
+      });
     }
-  }
 
-  public componentWillReceiveProps(newProps: IDocumentCardTitleProps): void {
-    this._events.off(window, 'resize');
-    this._isTruncated = false;
+    this._events.off(window, 'resize', this._updateTruncation);
```

Comment:
Why do we remove this and then add it right back? Shouldn't we only try to remove it if the newProps don't have truncate and the old props do?